### PR TITLE
Improve cluster creation and management docs

### PR DESCRIPTION
**Pull request by piontec**
This PR reorganizes mainly docs around how to create templates and instances of clusters.

Question for reviewers: what do you think about the `add_wc_structure.md` and `add_wc_instance.md` docs? Should we combine them into a single one? Leave as they are? Add an intro page that explains them both (like `add_wc_template.md`)?

### CODE REVIEW

File: docs/add_org.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Add a new organization
 
-Follow the below instructions to add a new organization to this repository.
+Follow the below instructions to add a new Organization to this repository. Organizations are created within the
+[Management Cluster's configuration](./add_mc.md) directory and in turn are meant to include
```
Comment (ljakimczuk) [inline docs/add_org.md:4]: isn't this enough to just say this instruction respects the repository structure? Structure is our contract I believe, so if we say we met this contract, it doesn't seem necessary to also mention this again here, wdyt?

File: docs/add_wc_instance.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+```sh
+export MC_NAME=CODENAME
+export ORG_NAME=ORGANIZATION
+export WC_NAME=CLUSTER_NAME
+```
+
+## Pick your Cluster Template
+
+First step needed to create a Cluster instance is to pick a cluster template it will be based on. Templates are available
+as Kustomize bases in the [bases](./bases/) directory. There's also
```
Comment (ljakimczuk) [inline docs/add_wc_instance.md:20]:
```suggestion
as Kustomize bases in the [bases](../bases/) directory. There's also
```

**Comment by ljakimczuk** [comment]
Looks good to me 👍 

In general, it is hard for me to decide the best structure around docs, too long docs are hard to follow, but on the other hand too many references and "jumps" between different docs feels also hard, but I'm in favour of trying out different approaches 😄

File: docs/add_org.md (hunk 1 above)
Comment (piontec) [reply docs/add_org.md:4]: Hard to say... My idea was that it's pretty important to point it out here explicitly than to hope that someone will click a URL instead of asking a question. I'll leave it, but add a remark that the complete description is in the 'structure' doc
